File: packages/langchain-basesentinel/src/basesentinel_langchain/summary.py
"""LLM-facing scan summary (no payment details)."""

from __future__ import annotations

from typing import Any


def summarize_scan_result(result: dict[str, Any]) -> str:
    address = result.get("address") or "unknown"
    status = result.get("status") or "UNKNOWN"
    verdict = result.get("verdict") or "UNKNOWN"
    score = result.get("verdict_score", "?")
    reasons = result.get("reasons")
    if isinstance(reasons, list):
        reason_text = ", ".join(str(r) for r in reasons if r) or "None"
    else:
        reason_text = "None"
    return (
        f"Contract {address} is {status} ({verdict}). "
        f"Score: {score}/100. Reasons: {reason_text}"
    )

File: packages/mcp-basesentinel/src/summary.ts
import type { ScanResult } from "./client.js";

/**
 * Short LLM-facing summary — no payment / tx details.
 */
export function summarizeScanResult(result: ScanResult): string {
  const address = result.address || "unknown";
  const status = result.status || "UNKNOWN";
  const verdict = result.verdict || "UNKNOWN";
  const score =
    typeof result.verdict_score === "number" ? result.verdict_score : "?";
  const reasons = Array.isArray(result.reasons)
    ? result.reasons.filter(Boolean)
    : [];
  const reasonText = reasons.length > 0 ? reasons.join(", ") : "None";
  const attribution =
    result.creator_attribution === "aa_bundler_mislabel"
      ? " Creator attribution: AA bundler mislabel (not a safe-token signal)."
      : "";

  return `Contract ${address} is ${status} (${verdict}). Score: ${score}/100. Reasons: ${reasonText}.${attribution}`;
}
